pygmt/src/subplot.py

```diff
@@ -0,0 +1,222 @@
+    r"""
+    Create multi-panel subplot figures.
+
+    The **subplot** module is used to split the current figure into a
+    rectangular layout of subplots that each may contain a single self-
+    contained figure. Begin by defining the layout of the entire multi-panel
+    illustration. Several options are available to specify the systematic
+    layout, labeling, dimensions, and more for the subplots.
+
+    Full option list at :gmt-docs:`subplot.html#synopsis-begin-mode`
+
+    {aliases}
+
+    Parameters
+    ----------
+    nrows : int
+        Number of vertical rows of the subplot grid.
+    ncols : int
+        Number of horizontal columns of the subplot grid.
+    figsize : tuple
+        Specify the final figure dimensions as ``(width, height)``.
+    subsize : tuple
+        Specify the dimensions of each subplot directly as ``(width, height)``.
+
+    autolabel : bool or str
+        [*autolabel*][**+c**\ *dx*\ [/*dy*]][**+g**\ *fill*][**+j**\|\ **J**\
+        *refpoint*][**+o**\ *dx*\ [/*dy*]][**+p**\ *pen*][**+r**\|\ **R**]
+        [**+v**].
+        Specify automatic tagging of each subplot. Append either a number or
+        letter [a]. This sets the tag of the first, top-left subplot and others
+        follow sequentially. Surround the number or letter by parentheses on
+        any side if these should be typeset as part of the tag. Use **+j|J**\
+        *refpoint* to specify where the tag should be placed in the subplot
+        [TL]. Note: **+j** sets the justification of the tag to *refpoint*
+        (suitable for interior tags) while **+J** instead selects the mirror
+        opposite (suitable for exterior tags). Append **+c**\ *dx*[/*dy*] to
+        set the clearance between the tag and a surrounding text box requested
+        via **+g** or **+p** [3p/3p, i.e., 15% of the :gmt-term:`FONT_TAG` size
+        dimension]. Append **+g**\ *fill* to paint the tag's text box with
+        *fill* [no painting]. Append **+o**\ *dx*\ [/*dy*] to offset the tag's
+        reference point in the direction implied by the justification [4p/4p,
+        i.e., 20% of the :gmt-term:`FONT_TAG` size]. Append **+p**\ *pen* to
+        draw the outline of the tag's text box using selected *pen* [no
+        outline]. Append **+r** to typeset your tag numbers using lowercase
+        Roman numerals; use **+R** for uppercase Roman numerals [Arabic
+        numerals]. Append **+v** to increase tag numbers vertically down
+        columns [horizontally across rows].
+    {B}
+    clearance : str
+        [*side*]\ *clearance*.
+        Reserve a space of dimension *clearance* between the margin and the
+        subplot on the specified side, using *side* values from **w**, **e**,
+        **s**, or **n**, or **x** for both **w** and **e** or **y** for both
+        **s** and **n**.  No *side* means all sides. The option is repeatable
+        to set aside space on more than one side. Such space will be left
```

Comment:
>  The option is repeatable to set aside space on more than one side.

This sentence doesn't make sense for PyGMT functions. We should rephrase it to a list of *clearance*.

Reply:
Ok, I'll rephrase it and add some examples in code-blocks.